### DISCUSSION TITLE
Highlight and annotation

### DISCUSSION THREAD

**Discussion by DD318** [sumatrapdfreader/sumatrapdf#4447] [Q&A]
Hi,
I have some problem with commenting tools. I have searched, but have not found the answer. 
1. is it possible to change the operation of the DEL key in pdf comments? Currently it works so that when I type a comment and press DEL it deletes the entire selection and comment without UNDO option. It is really frustrating to loss comments so easy.
2. is it possible to change the action of the note for the hatched text. Currently when you circle the text it is copied to the clipboard. I would like to avoid this.
3. I often work with files which have numbered lines. Most readers skip line numbering when selecting/ highlighting text in such documents. Sumatra selects everything that is in a line. Is there any way around this?

Kind regards,
DD318

**Comment by kjk** [comment]
1. is already fixed in [pre-release](https://www.sumatrapdfreader.org/prerelease)
2. Not sure what you're referring to. Please post a screenshot / more detailed description.
3. Also not sure, screenshot or sample document needed to understad

**Reply by DD318** [reply]
Ad. 1. 1 Thanks, I will check it at the earliest opportunity.

Ad. 2. Ok, my description was terrible. The point is that when I select a piece of text and highlight it, Sumatra copies that text to the clipboard. The problem is that I often already have text in the clipboard that I want to paste as a note to the highlighted piece. Sumatra then overwrites my cutout, with the text I highlighted, and I have to go out of the pdf and copy my texta gain. Of course, this can be solved by the correct order of operations, but that is not the point.
![copy-paste](https://github.com/user-attachments/assets/fef71600-d933-498e-9f36-61aa5f970ef6)


Ad. 3. This problem occurs in probably all pdf readers on Windows, so I do not know whether there is a point in fighting it. The point is that documents often have numbered lines and when you select a section of such a document, readers also mark the line numbering. In PDF Expert (IOS), the reader selects only the text and skips the line numbering. As a result, after copying you get only the text without line numbers. In Windows-based readers (Sumatra, among others) you get text with line numbers and then a line break mark, as in the screenshot above.
![row numbering](https://github.com/user-attachments/assets/2f8ebba3-91fd-41b1-ab4a-23e5e6a393bf)

**Reply by GitHubRulesOK** [reply]
@DD318 
the copy to clipboard is a default in some PDF applications but in newest SumatraPDF pre-release it is controllable without or clipped or clipped and copied those can be assigned to shortcuts/icons

My default is press R to highlight and open editor but not paste clipping
![image](https://github.com/user-attachments/assets/65416de0-d82f-4aad-b9db-d3ad71e8dd8d)
![image](https://github.com/user-attachments/assets/e44470ea-d4ba-409c-9da0-e51b6e56e669)

So my other clipboard contents from below are not replaced

>Arguments:
color : default, color
openedit : boolean, false if not given
copytoclipboard : boolean, false if not given. For highlight/underline/squiggly/strikeout annotations, copies the selection (text of annotation) to clipboard. This used to be default behavior for built-in a etc. keyboard shortcuts but now it has to be explicitly chosen.
setcontent : boolean, false if not give. For highlight/underline/squiggly/strikeout sets content of annotation to the selection (text of annotation)

Your second problem is more difficult as "numerals" invariably count as "line text content"

![image](https://github.com/user-attachments/assets/8c15e094-e138-4798-8a59-28e79e6268ba)
![image](https://github.com/user-attachments/assets/679f2cd7-d32f-4fd7-bf01-48f32d0a9a34)
